assets/react/v3/entries/course-builder/components/curriculum/question-types/MultipleChoiceAndOrdering.tsx

```diff
@@ -0,0 +1,242 @@
+import { colorTokens, spacing } from '@Config/styles';
+import { styleUtils } from '@Utils/style-utils';
+import { nanoid } from '@Utils/util';
+
+const MultipleChoiceAndOrdering = () => {
+  const isInitialRenderRef = useRef(false);
+  const [activeSortId, setActiveSortId] = useState<UniqueIdentifier | null>(null);
+  const form = useFormContext<QuizForm>();
+  const { activeQuestionIndex } = useQuizModalContext();
+  const {
+    fields: optionsFields,
+    append: appendOption,
+    insert: insertOption,
+    remove: removeOption,
+    move: moveOption,
+  } = useFieldArray({
+    control: form.control,
+    name: `questions.${activeQuestionIndex}.options`,
+  });
+  const sensors = useSensors(
+    useSensor(PointerSensor, {
+      activationConstraint: {
+        distance: 10,
+      },
+    }),
+    useSensor(KeyboardSensor, { coordinateGetter: sortableKeyboardCoordinates })
+  );
+
+  const currentOptions = useWatch({
+    control: form.control,
+    name: `questions.${activeQuestionIndex}.options`,
+    defaultValue: [],
+  });
+
+  const activeSortItem = useMemo(() => {
+    if (!activeSortId) {
+      return null;
+    }
+
+    return optionsFields.find((item) => item.ID === activeSortId);
+  }, [activeSortId, optionsFields]);
+
+  const hasMultipleCorrectAnswers = useWatch({
+    control: form.control,
+    name: `questions.${activeQuestionIndex}.multipleCorrectAnswer`,
+    defaultValue: false,
+  });
+
+  useEffect(() => {
+    isInitialRenderRef.current = true;
+    return () => {
+      isInitialRenderRef.current = false;
+    };
+  }, []);
+
+  // biome-ignore lint/correctness/useExhaustiveDependencies: <explanation>
+  useEffect(() => {
+    if (!hasMultipleCorrectAnswers && !isInitialRenderRef.current) {
+      const resetOptions = optionsFields.map((option) => ({ ...option, isCorrect: false }));
+      form.setValue(`questions.${activeQuestionIndex}.options`, resetOptions);
+    }
+    isInitialRenderRef.current = false;
+  }, [hasMultipleCorrectAnswers]);
+
+  // biome-ignore lint/correctness/useExhaustiveDependencies: <explanation>
+  useEffect(() => {
+    if (hasMultipleCorrectAnswers) {
+      return;
+    }
+
+    const changedOptions = currentOptions.filter((option) => {
+      const index = optionsFields.findIndex((item) => item.ID === option.ID);
+      const previousOption = optionsFields[index];
+      return option.isCorrect !== previousOption.isCorrect;
+    });
+
+    if (changedOptions.length === 0) {
+      return;
+    }
+
+    const changedOptionIndex = currentOptions.findIndex((item) => item.ID === changedOptions[0].ID);
+
+    const updatedOptions = [...currentOptions];
+    updatedOptions[changedOptionIndex] = Object.assign({}, updatedOptions[changedOptionIndex], { isCorrect: true });
+    updatedOptions.forEach((_, index) => {
+      if (index !== changedOptionIndex) {
+        updatedOptions[index] = Object.assign({}, updatedOptions[index], { isCorrect: false });
+      }
+    });
```

Comment:
Use `for...of` loop.
This code looks similar to some other code written before. You can use a utility function.